src/Compilers/CSharp/Portable/Parser/SyntaxFactoryContext.cs

```diff
@@ -24,6 +24,12 @@ internal class SyntaxFactoryContext
         /// </summary>
         internal bool IsInAsync;
 
+        /// <summary>
+        /// If we are forcing that ?[ is parsed as a conditional-access-expression, and not a conditional-expression
+        /// with a collection-expression in it.
+        /// </summary>
+        internal bool ForceConditionalAccessExpression;
```

Comment:
I'll try to formulate one. It would be very contrived though. Esp. because we do *not* reuse expressions. That's already because expressions *already* have a staggering amount of contextual parsing.  This is both because of things like speculative lookahead, and because of non-tracked context (like 'current expression precedence'). :-)